internal/nginx/config/generator.go

```diff
@@ -100,26 +104,46 @@ func generate(virtualServer state.VirtualServer, serviceStore state.ServiceStore
 		matches := make([]httpMatch, 0, len(rule.MatchRules))
 
 		for ruleIdx, r := range rule.MatchRules {
-
-			address, err := getBackendAddress(r.Source.Spec.Rules[r.RuleIdx].BackendRefs, r.Source.Namespace, serviceStore)
-			if err != nil {
-				warnings.AddWarning(r.Source, err.Error())
-			}
-
 			m := r.GetMatch()
 
+			var loc location
+
 			// handle case where the only route is a path-only match
 			// generate a standard location block without http_matches.
 			if len(rule.MatchRules) == 1 && isPathOnlyMatch(m) {
-				locs = append(locs, location{
-					Path:      rule.Path,
-					ProxyPass: generateProxyPass(address),
-				})
+				loc = location{
+					Path: rule.Path,
+				}
 			} else {
 				path := createPathForMatch(rule.Path, ruleIdx)
-				locs = append(locs, generateMatchLocation(path, address))
+				loc = generateMatchLocation(path)
 				matches = append(matches, createHTTPMatch(m, path))
 			}
+
+			// FIXME(pleshakov): There could be a case when the filter has the type set but not the corresponding field.
+			// For example, type is v1beta1.HTTPRouteFilterRequestRedirect, but RequestRedirect field is nil.
+			// The validation webhook catches that.
+			// If it doesn't work as expected, such situation is silently handled below in findFirstFilters.
+			// Consider reporting an error. But that should be done in a separate validation layer.
+
+			firstFilters := findFirstFilters(r.GetFilters(), supportedFilters)
+
+			loc.Return = generateReturnValForRedirectFilter(
```

Comment:
Can we easily extend this to support additional filters?


Why use the `supportedFilters` set to find the first filters but then assume support for redirect filter here?

Reply:
I think it would be easier to come up with a good design on supporting multiple filters once we implement more filters. So far, I just limited the extensibility to the  `findFirstFilters` function - you can add a new filter by changing the `supportedFilters`. However, the support for redirect filter doesn't include any design/code to support multiple ones (unless they are supported the same way -- adding more code to `generate` function)